plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php

```diff
@@ -163,32 +163,64 @@ function handleRevisionElement($node, $stageId, $fileId) {
 
 		$uploaderUsername = $node->getAttribute('uploader');
 		$uploaderUserGroup = $node->getAttribute('user_group_ref');
+		if (!$uploaderUsername) {
+			$user = $deployment->getUser();
+		} else {
+			// Determine the user based on the username
+			$userDao = DAORegistry::getDAO('UserDAO');
+			$user = $userDao->getByUsername($uploaderUsername);
+		}
+		if ($user) {
+			$submissionFile->setUploaderUserId($user->getId());
+		} else {
+			$deployment->addError(ASSOC_TYPE_SUBMISSION, $submission->getId(), __('plugins.importexport.common.error.unknownUploader', array('param' => $uploaderUsername)));
+			$errorOccured = true;
+		}
 
-		// Determine the user group based on the user_group_ref element.
+		// Determine the user group.
 		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
-		$userGroups = $userGroupDao->getByContextId($context->getId());
-		while ($userGroup = $userGroups->next()) {
-			if (in_array($uploaderUserGroup, $userGroup->getName(null))) {
-				$submissionFile->setUserGroupId($userGroup->getId());
-				break;
+		if ($user && !$uploaderUserGroup) {
+			// We have a user, but no group specified in the import.  Select a default group.
+			// We will prefer any group that has access to the submission's stage; find these
+			$stageUserGroups = $userGroupDao->getUserGroupsByStage($context->getId(), $stageId);
+			$stageUserGroupLookup = array();
+			while ($userGroup = $stageUserGroups->next()) {
+				$stageUserGroupLookup[] = $userGroup->getId();
+			}
+			// Check all user groups from this user
+			$userUserGroups = $userGroupDao->getByUserId($user->getId(), $context->getId());
+			$lastUserGroup = null;
+			while ($userGroup = $userUserGroups->next()) {
+				// If the user's user group has access to this stage, select it.
+				$lastUserGroup = $userGroup->getId();
+				if (in_array($lastUserGroup, $stageUserGroupLookup, true)) {
+					$submissionFile->setUserGroupId($userGroup->getId());
+					break;
+				}
+				// No matching user group has access to this stage, select the last one seen.
+				if ($lastUserGroup) {
```

Comment:
This should have been outside the while in case the user had no role in the context.